lib/chef/provider/service/insserv.rb

```diff
@@ -36,7 +36,7 @@ def load_current_resource
           super
 
           # Look for a /etc/rc.*/SnnSERVICE link to signify that the service would be started in a runlevel
-          if Dir.glob("/etc/rc**/S*#{Chef::Util::PathHelper.escape_glob_dir(current_resource.service_name)}").empty?
+          if Dir.glob("/etc/rc.d/rc**/S*#{Chef::Util::PathHelper.escape_glob_dir(current_resource.service_name)}").empty?
```

Comment:
This may fix it for SLES 11 but will almost certainly break it for other operating systems that used the `/etc/rc**/S*` pattern (RHEL,CentOS). I think we may need a better regex or some logic to determine the OS and version and select the correct regex accordingly.

Reply:
Thanks @devoptimist. I am investigating for other OSs and will update accordingly.

Reply:
This PR is updated and `action: enable` is idempotent for RHEL/CentOS as well as for SLESS11 now.
Trying to write some functional test cases now.

Reply:
I have verified there is no `/sbin/insserv` file on `RHEL-6/CentOs-6` and on `Ubuntu-14.04` so currently applied filter `sles11` in functional tests.